### PR TITLE
Rwa-blogs: add devise gem

### PR DESCRIPTION
**Pull request by BertrandConxy**
## Rwa-blogs: add devise gem

Hello ✋,
In this branch, I have done the following:
### Tasks completed
- Installed devise gem
- Configured devise with User module
- Customised sign in and sign up pages
- Added profile tab for a current user
- User can reset his/her password

### CODE REVIEW

File: db/migrate/20220515110719_add_devise_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
```
Comment (SajjadAhmad14) [inline db/migrate/20220515110719_add_devise_to_users.rb:1]: There is a migration error so I was not able to run the project. Kindly fix the issue.
devise

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (SajjadAhmad14) [inline app/models/user.rb]: Please add `:confirmable` to `devise` so the app is able to confirm credentials.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Bertrand, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/migrate/20220515110719_add_devise_to_users.rb (hunk 1 above)
Comment (BertrandConxy) [reply db/migrate/20220515110719_add_devise_to_users.rb:1]: Thank you @SajjadAhmad14 for your reviews. After realizing that this migration that caused such errors doesn't do anything important, I solved the issue by deleting it. I tested by running migrations and checking my schema.rb file and I found that everything is good now.

File: app/models/user.rb (hunk 2 above)
Comment (BertrandConxy) [reply app/models/user.rb]: Thank you again for your reviews. I have added this `:confirmable` attribute in the users model.

**Comment by BertrandConxy** [comment]
I have implemented all the changes required. I thereby look for the next review.

**Review by elmar8287 — APPROVED** [review]
Hi, 

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by BertrandConxy** [comment]
Thank you very much

**Pull request merged**